1.14/test/run

```diff
@@ -339,65 +350,68 @@ test_perl_directive() {
 
   info "Testing perl_set directive sets header with perl version"
   test_for_output "http://$(container_ip):${test_port}/" "X-Perl-Version: ${perl_version}"
-  check_result $? "perl-test-app"
+  check_result $?
 
   info "Testing perl directive runs perl location handler"
   test_for_output "http://$(container_ip):${test_port}/perl" "Perl location handler is working"
-  check_result $? "perl-test-app"
+  check_result $?
 
   cleanup_test_app
 }
 
-cid_file=$(mktemp -u --suffix=.cid)
-
-# Since we built the candidate image locally, we don't want S2I attempt to pull
-# it from Docker hub
-s2i_args="--pull-policy=never"
-
-prepare "test-app"
-run_s2i_build "test-app"
-check_result $? "test-app"
+test_application_user() {
+  # Test application with random uid
+  CONTAINER_ARGS="--user 12345" test_application
+}
 
-# Verify the 'usage' script is working properly when running the base image with 's2i usage ...'
-test_s2i_usage
-check_result $? "test-app"
 
-# Verify the 'usage' script is working properly when running the base image with 'docker run ...'
-test_docker_run_usage
-check_result $? "test-app"
+function run_all_tests() {
+  for test_case in $TEST_LIST; do
+    info "Running test $test_case ...."
+    if $test_case; then
+      printf -v test_short_summary "${test_short_summary}[PASSED] $test_case\n"
+    else
+      printf -v test_short_summary "${test_short_summary}[FAILED] $test_case\n"
+      [ -n "${FAIL_QUICKLY:-}" ] && echo "$sum" && return 1
+    fi
+  done;
+  cleanup "${APP_NAME}"
+}
 
-# Test application with default uid
-test_application
+function build_image() {
+  local test_name="$1"
+  cid_file=$(mktemp -u --suffix=.cid)
```

Comment:
Not sure if the cid file should be defined once over here, during the build of the image as it is used for runtime containers.

Reply:
Though it seems like each test case is doing its cidfile cleanup when necessary, so might be ok.